lib/puppet/parser/scope.rb

```diff
@@ -871,9 +871,12 @@ def to_s
 
     # shorten the path if possible
     path = detail[0]
-    if environment && environment.configuration && path.start_with?(environment.configuration.path_to_env)
-      path = "<env>" + path[environment.configuration.path_to_env.length..-1]
+    env_path = environment && environment.configuration && environment.configuration.path_to_env
```

Comment:
This expression makes absolutely no sense to me. Can you please rewrite it without using boolean operators on strings?

Reply:
!environment.nil? && !environment.configuration.nil? && !environment.configuration.path_to_env.nil?